### PR TITLE
Updated Spark setup script

### PR DESCRIPTION
**Pull request by vijaypavann-db**
1. Updated Spark setup script to check whether `spark gzip file` exists

1. Refactored the script to remove the warnings: 
    * `Double quote to prevent globbing and word splitting`
    * `Use 'cd ... || exit' or 'cd ... || return' in case cd fails.`
    * `Consider using 'grep -c' instead of 'grep|wc -l'.`
    * `Usage: sleep seconds`, converted 2m to 120 seconds

1. Tested the following Scenarios:
    1. Scenario: Extracted Spark folder is already present
        Outcome: Directly starts the spark server using `sbin/start-connect-server.sh`
    1. Scenario: Extracted Spark folder **is not present**, and Zip file (`spark-<VERSION>.tgz`) is present
        Outcome:  Extract the zip file and start the spark server
    1. Scenario: Extracted Spark folder **is not present**, and Zip file **is not present**
        Outcome:  Download, Extract and start the spark server

### CODE REVIEW

File: .github/scripts/setup_spark_remote.sh [hunk 1]
```diff
@@ -2,31 +2,36 @@
 
 set -xv
```
Comment (bishwajit-db) [inline .github/scripts/setup_spark_remote.sh:3]: Consider also adding -e to exit on any command failure

File: .github/scripts/setup_spark_remote.sh [hunk 2]
```diff
@@ -2,31 +2,36 @@
...
 
-mkdir -p $HOME/spark
-cd $HOME/spark
+mkdir -p "$HOME"/spark
+cd "$HOME"/spark || exit
 
 version=$(wget -O - https://dlcdn.apache.org/spark/ | grep 'href="spark' | grep -v 'preview' | sed 's:</a>:\n:g' | sed -n 's/.*>//p' | tr -d spark- | tr -d / | sort -r --version-sort | head -1)
```
Comment (bishwajit-db) [inline .github/scripts/setup_spark_remote.sh:8]: Add error checking after the version assignment. For, example,
if [ -z "$version" ]; then echo "Failed to extract Spark version"; exit 1; fi

File: .github/scripts/setup_spark_remote.sh [hunk 3]
```diff
@@ -2,31 +2,36 @@
...
 else
-  wget https://dlcdn.apache.org/spark/spark-${version}/${spark}.tgz
-  tar -xvf ${spark}.tgz
+  if [ -f "${spark}.tgz" ];then
+    echo "${spark}.tgz already exists"
+  else
+    wget "https://dlcdn.apache.org/spark/spark-${version}/${spark}.tgz"
+  fi
+  tar -xvf "${spark}.tgz"
 fi
```
Comment (bishwajit-db) [inline .github/scripts/setup_spark_remote.sh:27]: Should the script clean up the downloaded .tgz file after extraction?

File: .github/scripts/setup_spark_remote.sh [hunk 4]
```diff
@@ -2,31 +2,36 @@
 
 set -xv
 
-mkdir -p $HOME/spark
-cd $HOME/spark
+mkdir -p "$HOME"/spark
+cd "$HOME"/spark || exit
```
Comment (bishwajit-db) [inline .github/scripts/setup_spark_remote.sh:6]: Exit with non-zero status if cd fails. Like exit 1. Same for other places.

**Review by bishwajit-db — CHANGES_REQUESTED** [review]

File: .github/scripts/setup_spark_remote.sh (hunk 3 above)
Comment (vijaypavann-db) [reply .github/scripts/setup_spark_remote.sh:27]: Keeping it to avoid downloading every time

File: .github/scripts/setup_spark_remote.sh (hunk 1 above)
Comment (vijaypavann-db) [reply .github/scripts/setup_spark_remote.sh]: Done

File: .github/scripts/setup_spark_remote.sh (hunk 2 above)
Comment (vijaypavann-db) [reply .github/scripts/setup_spark_remote.sh:8]: done

File: .github/scripts/setup_spark_remote.sh (hunk 4 above)
Comment (vijaypavann-db) [reply .github/scripts/setup_spark_remote.sh]: done

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Review by bishwajit-db — APPROVED** [review]
LGTM

**Pull request merged**